components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2347,23 +2320,50 @@ bool MaybeTailDiagnosticLog(
 
   first_run = false;
 
-  if (!TailFile(&g_diagnostic_log, num_lines)) {
+  if (!TailFile(&diagnostic_log_, num_lines)) {
     return false;
   }
 
   return true;
 }
 
-bool WriteToDiagnosticLogOnFileTaskRunner(
+void RewardsServiceImpl::DiagnosticLog(
+    const std::string& file,
+    const int line,
+    const int verbose_level,
+    const std::string& message) {
+  if (profile_->IsOffTheRecord()) {
+    return;
+  }
+
+  if (verbose_level > kDiagnosticLogMaxVerboseLevel) {
+    return;
+  }
+
+  base::PostTaskAndReplyWithResult(file_task_runner_.get(), FROM_HERE,
+      base::BindOnce(&RewardsServiceImpl::WriteToDiagnosticLogOnFileTaskRunner,
+          base::Unretained(this),
+          diagnostic_log_path_,
+          kTailDiagnosticLogToNumLines,
+          file,
+          line,
+          verbose_level,
+          message),
+      base::BindOnce(&RewardsServiceImpl::OnWriteToLogOnFileTaskRunner,
+          AsWeakPtr()));
+}
+
+
+bool RewardsServiceImpl::WriteToDiagnosticLogOnFileTaskRunner(
     const base::FilePath& log_path,
```

Comment:
just a nit - there's really no reason to pass the log_path here because you can access it directly from `diagnostic_log_path_`